src/hooks/useNote.js

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react'
+
+function useNote({ serviceClient, noteId }) {
+  // `docId` is the id of the note for which we have a state.
+  // `noteId` is the id of the requested note.
+  const [docId, setDocId] = useState(noteId)
+  const [loading, setLoading] = useState(true)
+  const [doc, setDoc] = useState(undefined)
+  const [title, setTitle] = useState(undefined)
+
+  // reload if ever noteId changes
+  useEffect(
+    () => {
+      if (docId !== noteId) {
+        setLoading(true)
+        setTitle(undefined)
+        setDoc(undefined)
+        setDocId(noteId)
+      }
+    },
+    [noteId, docId, setLoading, setTitle, setDoc, setDocId]
+  )
+
+  // load the note
+  useEffect(
+    () => {
+      const loadNote = async function() {
+        try {
+          if (!loading) setLoading(true)
+          const doc = await serviceClient.getDoc(docId)
+          setTitle(doc.title || '')
+          setDoc(doc)
+        } catch (e) {
+          setTitle(false)
+          setDoc(false)
+          // eslint-disable-next-line no-console
+          console.warn(`Could not load note ${docId}`)
+        }
+        setLoading(false)
+      }
+      if (serviceClient) {
+        if (!doc || doc.file.id != docId) loadNote()
+      }
+    },
+    [docId, setLoading, serviceClient, setDoc, setTitle]
+  )
+
+  if (docId == noteId) return { loading, title, doc, setTitle }
+  else return { loading: true, title: undefined, doc: undefined, setTitle }
```

Comment:
Aren't these values set though the first `useEffect` anyway?

Reply:
nope.

- You run the `useNote({noteId: 1})`, load the note 1, fill `title`, `doc` and `loading` for `docId=1`.
- Then you run `useNote({noteId: 2})` in a future render.  The effect is async and won't execute immediatly (it will be planned just after the render of the component calling `useNote`). You will then reach the end of `useNote` with noteId=2, docId=1, and the title/doc/loading values of docId=1, which is not  what the calling component has asked.

Once you have an effect, you should always think that your data may lag of one render, and add safeguards to avoid filling obsolete values

Reply:
Mmh ok. I think the `noteId`/`docId` is a bit confusing and there might be a better wya to handle this, but it wasn't the scope of the PR and it's already easier to understand things with asingle hook, so LGTM 👍